src/yukon/backend/src/api/controllers/nodes/__init__.py

```diff
@@ -1,129 +1,256 @@
+                                                    990, '4.3.2.1', '0xTOOMUCHBEEF', '1.2.3.4', 'my-awesome-uid',
+                                                    'I am authentic')
+
+            return jsonify(mock_responses.serialise())
+
+        @self.nodes_controller.route('/<int:nodeId>/shutdown', methods=['PUT'])
+        async def node_shutdown(nodeId) -> Tuple[Response, int]:
+            return "", 200
+
+        @self.nodes_controller.route('/<int:nodeId>/firmwareupdate', methods=['PUT'])
+        async def node_firmware_update(nodeId, param) -> Tuple[Response, int]:
+            body = await request.get_json()
+            return body['name'], 200
+
+    @property
+    def nodes_controller(self):
+        return self._nodes_controller
+
+    @property
+    def node_list(self):
+        return self._node_list
+
+    @property
+    def nodeid_list(self):
+        return self._nodeid_list
+
+    @classmethod
+    def health_to_text(self, health_code):
+        node_health_code = {uavcan.node.Heartbeat_1_0.HEALTH_NOMINAL: 'NOMINAL',
+                            uavcan.node.Heartbeat_1_0.HEALTH_ADVISORY: 'ADVISORY',
+                            uavcan.node.Heartbeat_1_0.HEALTH_CAUTION: 'CAUTION',
+                            uavcan.node.Heartbeat_1_0.HEALTH_WARNING: 'WARNING'}
+        return node_health_code.get(health_code)
+
+    @classmethod
+    def mode_to_text(self, mode_code):
+        node_mode_code = {uavcan.node.Heartbeat_1_0.MODE_OPERATIONAL: 'OPERATIONAL',
+                          uavcan.node.Heartbeat_1_0.MODE_INITIALIZATION: 'INITIALIZATION',
+                          uavcan.node.Heartbeat_1_0.MODE_MAINTENANCE: 'MAINTENANCE',
+                          uavcan.node.Heartbeat_1_0.MODE_SOFTWARE_UPDATE: 'SOFTWARE_UPDATE',
+                          uavcan.node.Heartbeat_1_0.MODE_OFFLINE: 'OFFLINE'}
+        return node_mode_code.get(mode_code)
+
+
+class NodeInfo(object):
+    """
+        Class strucuture for UAVCAN node info.
+    """
+
+    def __init__(self, name: str, id: int, health: str, mode: str, uptime: int, vendor_code: int) -> None:
+        self.name = name
+        self.id = id
+        self.health = health
+        self.mode = mode
+        self.uptime = uptime
+        self.vendor_code = vendor_code
+
+    def serialise(self) -> Dict[str, Any]:
+        return {
+            'name': self.name,
+            'id': self.id,
+            'health': self.health,
+            'mode': self.mode,
+            'uptime': self.uptime,
+            'vendorCode': self.vendor_code
+        }
+
+
+class Monitor(Controller):
+    """
+        UAVCAN node created in a specific bus, which purpose is to gather data from other existing
+        nodes and retrieve data from the bus itself, the nodes and the packets being exchanged.
+    """
+
+    def __init__(self):
+        Controller.__init__(self)
+        asyncio.get_event_loop().create_task(self._monitor_run())
```

Comment:
this line is strange. You create a task in the constructor but then you don't save a reference to the task. Mainly I don't like this because constructors shouldn't have side effects (ideally). Let's create a `def start()` method that kicks off the task. It also helps for debugging to save a reference to the task on the object.

```
def start(self) -> None:
    if self._task is None:
        self._task = asyncio.get_event_loop().create_task(self._monitor_run())
```